backend/src/modules/Boards/service/Boards.service.js

```diff
@@ -54,4 +55,23 @@ export class BoardsService {
     const updatedBoard = await this.repository.addNewAnnouncementId(boardId, announcementId);
     return updatedBoard;
   }
+
+  async getBoardAnnouncementsList(boardId) {
+    if (!mongoose.isValidObjectId(boardId) && this.repository.constructor.name == 'MONGO') {
```

Comment:
`this.repository.constructor.name == 'MONGO'`
Tym nie powinien zajmować się Service.
Service nie powinien mieć wiedzy czy to  MONGO czy SQL czy InMemory czy coś innego :)
Jeśli problemem jest ID, to przydałoby się rozwiązać ten problem z ID (np. nadając go w Servicie, a nie w repo) - tak strzelam, bo nie wiem, czy dlatego to zmieniłaś.

Reply:
Oooo, jeszcze jedno. Teraz chyba widzę.
Jeśli io jest ObjectId (ale nie stworzony przez ciebie w ValueObject), tylko pochodzący z mongo, to nie możemy tego tak zostawić. Repo nie może zwracać Mongo ObjectId. Może co najwyżej zwracać, samo ID ;)

Reply:
Jak coś to możesz to zmargować i naprawić w kolejnym PR :)

Reply:
W sumie to nie potrzebuję tej linii. Mongoose sam wyrzuci błąd jeśli będzie błędny typ id.